### ISSUE TITLE
regression importing cygrpc grpc python 1.24.0

### ISSUE THREAD

**Issue by jkohen** [grpc/grpc#20404]
### What version of gRPC and what language are you using?
1.24.0 python

### What operating system (Linux, Windows,...) and version?
Windows

### What runtime / compiler are you using (e.g. python version or version of gcc)
Python 3.6.5 from https://www.python.org/ftp/python/3.6.5/python-3.6.5.exe

### What did you do?
If possible, provide a recipe for reproducing the error. Try being specific and include code snippets if helpful.
Imported the google-cloud-monitoring 0.33.0 library.

### What did you expect to see?
My test runs.

### What did you see instead?
```python
my_test.py:7: in <module>
    from google.cloud import monitoring
Python36\lib\site-packages\google\cloud\monitoring.py:20: in <module>
    from google.cloud.monitoring_v3 import AlertPolicyServiceClient
Python36\lib\site-packages\google\cloud\monitoring_v3\__init__.py:21: in <module>
    from google.cloud.monitoring_v3.gapic import alert_policy_service_client
Python36\lib\site-packages\google\cloud\monitoring_v3\gapic\alert_policy_service_client.py:26: in <module>
    import google.api_core.gapic_v1.config
Python36\lib\site-packages\google\api_core\gapic_v1\__init__.py:16: in <module>
    from google.api_core.gapic_v1 import config
Python36\lib\site-packages\google\api_core\gapic_v1\config.py:23: in <module>
    import grpc
Python36\lib\site-packages\grpc\__init__.py:23: in <module>
    from grpc._cython import cygrpc as _cygrpc
E   ImportError: DLL load failed: The specified module could not be found.
```

Make sure you include information that can help us debug (full error message, exception listing, stack trace, logs).

See [TROUBLESHOOTING.md](https://github.com/grpc/grpc/blob/master/TROUBLESHOOTING.md) for how to diagnose problems better.

### Anything else we should know about your project / environment?
This was working with grpcio-1.23.0. I successfully worked around the issue with 1.24.0 by upgrading to Python 3.7.4 from https://www.python.org/ftp/python/3.7.4/python-3.7.4.exe

**Comment by lidizheng** [comment]
@jkohen This failure seems due to failed installation of gRPC Python package. It might happen when your pip install doesn't pick up the right binary wheel, also failed to compile it locally. Can you try to install `grpcio` manually? And paste the output?

**Comment by jkohen** [comment]
@lidizheng I'm sorry, I don't think I will be able to test that. I'm not  familiar enough with the environment and with Python to effectively do that.

**Comment by lidizheng** [comment]
Several updates:
1. It works fine for normal GCP Windows VM;
1. Regression confirmed on the image provided (1.23.0 works fine, and 1.24.0 cannot load DLL);
2. The image is a 64 bit Windows, but the CPython you are using is 32 bit. Please consider replace it with "https://www.python.org/ftp/python/3.6.5/python-3.6.5-amd64.exe".
3. It seems recent C-Core update did change the dependencies, but needs further investigation.
    * After install the investigation tool, I accidentally picked up a newer VCRUNTIME...

```
# List of depdencies
ADVAPI32.dll
WS2_32.dll
python36.dll
KERNEL32.dll
VCRUNTIME140.dll
api-ms-win-cht-heap-l1-1-0.dll
api-ms-win-cht-runtime-l1-1-0.dll
api-ms-win-cht-utility-l1-1-0.dll
api-ms-win-cht-stdio-l1-1-0.dll
api-ms-win-cht-time-l1-1-0.dll
api-ms-win-cht-string-l1-1-0.dll
api-ms-win-cht-convert-l1-1-0.dll
api-ms-win-cht-math-l1-1-0.dll
api-ms-win-cht-filesystem-l1-1-0.dll
api-ms-win-cht-environment-l1-1-0.dll
```

![image](https://user-images.githubusercontent.com/7394928/66010228-eae82d80-e472-11e9-89d1-06c01f76e592.png)

**Comment by lidizheng** [comment]
Another update about this issue:

Turn out the binary wheel for 3.6 depends on "MSVCP140.DLL" which should be linked statically in the first place. @veblush has a fix in #20451 to address this problem. Also, Python 3.5 has the same problem as well.

We also confirmed that 3.7 is unaffected, due to ... unknown reason. Python 3.4 is fine since we are using another compiler.

| Python Version | Compiler | DLL load |
|---|---|---|
| 3.4 | mingw32 | Success |
| 3.5 | msvc | Failed |
| 3.6 | msvc | Failed |
| 3.7 | msvc | Success |

Dependencies of 1.24 on Python 3.6 (3.5 is similar):
![image](https://user-images.githubusercontent.com/7394928/66080096-3bff2c80-e51a-11e9-8532-fc727badbfa0.png)

Dependencies of 1.24 on Python 3.4:
![image](https://user-images.githubusercontent.com/7394928/66080672-6dc4c300-e51b-11e9-9356-826a78723b6e.png)
